### PR TITLE
[Locale] Locale refactoring

### PR DESCRIPTION
**Pull request by pamil**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | yes
| BC breaks?      | yes
| Related tickets | -
| License         | MIT

Basically the same as currency refactoring in #5497, but much simplier :)

### CODE REVIEW

**Comment by pjedrzejewski** [comment]
Thanks Kamil! 👍

File: src/Sylius/Bundle/LocaleBundle/Listener/RequestLocaleSetter.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    {
+        $request = $event->getRequest();
+
+        try {
+            $request->setLocale($this->localeContext->getLocaleCode());
+        } catch (LocaleNotFoundException $exception) {}
+
+        try {
+            $request->setDefaultLocale($this->localeProvider->getDefaultLocaleCode());
+        } catch (LocaleNotFoundException $exception) {}
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/LocaleBundle/Listener/RequestLocaleSetter.php]: Default locale is mandatory, so we shouldn't silent the exception.

Comment (pamil) [inline src/Sylius/Bundle/LocaleBundle/Listener/RequestLocaleSetter.php]: Had some reasoning about it, but can't recall it. I'll take a look & think how to manage this in #5601.

**Comment by ylastapis** [comment]
Hi @pamil 

Could you tell me now how to set the currentLocale as LocaleContext became ImmutableLocaleContext ? :)

My requested locale came from the _locale parameter of my route (API calls)
Maybe there's a better way to do that...

Thanks !

**Pull request merged**